src/core/tests/Test_TelemetryWriter.py

```diff
@@ -102,19 +102,19 @@ def test_write_event_msg_size_limit(self):
             f.close()
 
     def test_write_event_msg_size_limit_char_more_than_1_bytes(self):
-        """ Perform 1 byte truncation on char that is more than 1 byte, use decode('utf-8', errors='replace') to replace bad unicode with a good 1 byte char (�) """
+        """ Perform 1 byte truncation on char that is more than 1 byte, use decode('utf-8', errors='replace') to replace bad unicode with a good 1 byte char (\uFFFD) """
 
-        message = "a€bc"*3074  # €(\xe2\x82\xac) is 3 bytes char can be written in windows console w/o encoding
+        message = "a\u20acbc" * 3074  # (\xe2\x82\xac) is 3 bytes char can be written in windows console w/o encoding
         self.runtime.telemetry_writer.write_event(message, Constants.TelemetryEventLevel.Error, "Test Task")
         latest_event_file = [pos_json for pos_json in os.listdir(self.runtime.telemetry_writer.events_folder_path) if re.search('^[0-9]+.json$', pos_json)][-1]
         with open(os.path.join(self.runtime.telemetry_writer.events_folder_path, latest_event_file), 'r+') as f:
             events = json.load(f)
             self.assertTrue(events is not None)
             self.assertEqual(events[-1]["TaskName"], "Test Task")
             self.assertTrue(len(events[-1]["Message"]) < len(message.encode('utf-8')))
-            chars_dropped = len(message.encode('utf-8')) - Constants.TELEMETRY_MSG_SIZE_LIMIT_IN_CHARS + Constants.TELEMETRY_BUFFER_FOR_DROPPED_COUNT_MSG_IN_CHARS + Constants.TELEMETRY_EVENT_COUNTER_MSG_SIZE_LIMIT_IN_CHARS
-            self.assertTrue("a€bc" in events[-1]["Message"])
-            self.assertTrue("a€bc" * (len(message) + 1 - chars_dropped) + ". [{0} chars dropped]".format(chars_dropped) in events[-1]["Message"])  # len(message) + 1 due to bad unicode will be replaced by �
```

Comment:
Why were the additional assertions removed?

Reply:
@feng-j678 this is still open

Reply:
@feng-j678 please update what was discussed between you and Koshy about this feedback here. This gives me clarity and will also be useful in future

Reply:
Copying reasoning from PR description

<img width="562" alt="image" src="https://github.com/user-attachments/assets/9ca333a3-7d47-4570-ae82-553e2a4fb7e0">